packages/elasticsearch-plugin/src/elasticsearch-resolver.ts

```diff
@@ -62,17 +53,25 @@ export class AdminElasticSearchResolver implements SearchResolver {
         return result;
     }
 
-    @ResolveField()
-    async facetValues(
-        @Ctx() ctx: RequestContext,
-        @Parent() parent: { input: SearchInput },
-    ): Promise<Array<{ facetValue: FacetValue; count: number }>> {
-        return this.elasticsearchService.facetValues(ctx, parent.input, false);
-    }
-
     @Mutation()
     @Allow(Permission.UpdateCatalog)
     async reindex(@Ctx() ctx: RequestContext): Promise<GraphQLJob> {
         return (this.elasticsearchService.reindex(ctx, false) as unknown) as GraphQLJob;
     }
 }
+
+@Resolver('SearchResponse')
+export class EntityElasticSearchResolver implements Pick<SearchResolver, 'facetValues'> {
+    constructor(private facetValueService: FacetValueService) {}
+
+    @ResolveField()
+    async facetValues(
+        @Ctx() ctx: RequestContext,
+        @Parent() parent: Omit<SearchResponse, 'facetValues'>,
+    ): Promise<Array<{ facetValue: FacetValue; count: number }>> {
+        const facetValueIds = parent.items.map(item => item.facetValueIds).flat();
```

Comment:
I believe this change to the way facetValues are counted is responsible for some of the test failures. This change does not replicate the old behaviour: previously, it would return the facetValues counts for the _entire_ result set. Now it is only returning the count for the current page of results.

Reply:
I see. We are running this revised plugin for quite some time now but hadn't noticed as we only use this for the items in the current result set. Now I also understand why there were two calls to the Elastic instance before. Should be relatively straightforward to put the old system back in place.

Reply:
Alright, I'll revert that part 👍